symbolic-sourcemapcache/src/sourcemapcache/writer.rs

```diff
@@ -191,75 +188,51 @@ impl SourceMapCacheWriter {
         let files = files.into_iter().map(|(_name, file)| file).collect();
 
         Ok(Self {
-            string_bytes,
+            string_table,
             files,
             line_offsets,
             mappings,
         })
     }
 
-    /// Insert a string into this converter.
-    ///
-    /// If the string was already present, it is not added again. A newly added string
-    /// is prefixed by its length in LEB128 encoding. The returned `u32`
-    /// is the offset into the `string_bytes` field where the string is saved.
-    fn insert_string(
-        string_bytes: &mut Vec<u8>,
-        strings: &mut HashMap<String, u32>,
-        s: &str,
-    ) -> u32 {
-        if s.is_empty() {
-            return u32::MAX;
-        }
-        if let Some(&offset) = strings.get(s) {
-            return offset;
-        }
-        let string_offset = string_bytes.len() as u32;
-        let string_len = s.len() as u64;
-        leb128::write::unsigned(string_bytes, string_len).unwrap();
-        string_bytes.extend(s.bytes());
-
-        strings.insert(s.to_owned(), string_offset);
-        string_offset
-    }
-
     /// Serialize the converted data.
     ///
     /// This writes the SourceMapCache binary format into the given [`Write`].
     #[tracing::instrument(level = "trace", name = "SourceMapCacheWriter::serialize", skip_all)]
     pub fn serialize<W: Write>(self, writer: &mut W) -> std::io::Result<()> {
-        let mut writer = WriteWrapper::new(writer);
+        let mut writer = Writer::new(writer);
+        let string_bytes = self.string_table.as_bytes();
```

Comment:
This reminds me: the `StringTable` could use a `into_bytes(self) -> Vec<u8>` which just drops the hashtable early if you don’t need it anymore.

Reply:
Good idea, I'll add it.